test/core/resampling/test_affine.py

```diff
@@ -44,6 +46,91 @@
 
 class AffineTransformDatasetTest(unittest.TestCase):
     def test_subset(self):
+        target_gm = GridMapping.regular((3, 3), (50.0, 10.0), res, source_gm.crs)
+        target_ds = affine_transform_dataset(
+            source_ds, source_gm, target_gm, gm_name="crs"
+        )
+        self.assertIsInstance(target_ds, xr.Dataset)
+        self.assertEqual(
+            set(source_ds.variables).union(["crs"]), set(target_ds.variables)
+        )
+        self.assertEqual((3, 3), target_ds.refl.shape)
+        np.testing.assert_almost_equal(
+            target_ds.refl.values,
+            np.array(
+                [
+                    [1, 0, 2],
+                    [0, 3, 0],
+                    [4, 0, 1],
+                ]
+            ),
+        )
+
+        target_gm = GridMapping.regular((3, 3), (50.1, 10.1), res, source_gm.crs)
+        target_ds = affine_transform_dataset(
+            source_ds, source_gm=source_gm, target_gm=target_gm, gm_name="crs"
+        )
+        self.assertIsInstance(target_ds, xr.Dataset)
+        self.assertEqual(
+            set(source_ds.variables).union(["crs"]), set(target_ds.variables)
+        )
+        self.assertEqual((3, 3), target_ds.refl.shape)
+        np.testing.assert_almost_equal(
+            target_ds.refl.values,
+            np.array(
+                [
+                    [4, 0, 1],
+                    [0, 2, 0],
+                    [3, 0, 4],
+                ]
+            ),
+        )
+
+        target_gm = GridMapping.regular((3, 3), (50.05, 10.05), res, source_gm.crs)
+        target_ds = affine_transform_dataset(
+            source_ds, source_gm=source_gm, target_gm=target_gm, gm_name="crs"
+        )
+        self.assertIsInstance(target_ds, xr.Dataset)
+        self.assertEqual(
+            set(source_ds.variables).union(["crs"]), set(target_ds.variables)
+        )
+        self.assertEqual((3, 3), target_ds.refl.shape)
+        np.testing.assert_almost_equal(
+            target_ds.refl.values,
+            np.array([[1.25, 1.5, 0.75], [1.0, 1.25, 1.5], [1.75, 1.0, 1.25]]),
```

Comment:
formatting as the previous test, which makes it easier to read.

Reply:
Ok.